generator-java/generators/app/templates/microservice/liberty/src/main/java/application/rest/RootEndpoint.java

```diff
@@ -19,4 +24,14 @@ public Response listResources(@Context UriInfo uriInfo) {
       return Response.ok("{\"health\":\""+healthURL+"\",\"example\":\""+exampleURL+"\"}").build();
     }
 
+    @GET
+    @Produces({MediaType.TEXT_HTML})
+    public InputStream getIndex() {
+      try {
+        File f = new File("../../../../../classes/index.html");
+        return new FileInputStream(f);
+      } catch (FileNotFoundException e) {
+        return null;
+      }
+    }
 }
```

Comment:
Quan, I think there's a cleaner way to get the index.html input stream. See line 59 of:

https://github.com/ibm-developer/generator-ibm-service-enablement/blob/master/generators/language-java/templates/java-liberty/src/main/java/application/ibmcloud/CloudServices.java
```java
InputStream fstream = this.getClass().getResourceAsStream(path);
```
In that file `path` is either `"/mappings.json"` or `"/localdev-config.json"` which are in the same dir as index.html, so `/"index.html"` should work.

But that should require that index.html be in resources dir rather than webapp I believe.

Reply:
Cool! I suspected so. Thanks I'll try it out.